### PR TITLE
Environment variable as a prefix for all etcd records. (#8567)

### LINKED ISSUE #8567: Environment variable as a prefix for all etcd records.
**Is your feature request related to a problem? Please describe.**

We can't configure a prefix for all keys stored on etcd. As a result, we can't use an existing etcd cluster and need to create another only for MinIO 😞 

**Describe the solution you'd like**

Add environment variable `MINIO_ETCD_PREFIX` and use it as a prefix for all etcd records.

**Describe alternatives you've considered**

Create and **support** yet another etcd instance.

### PR DESCRIPTION
**Pull request by ihostage**
## Motivation and Context

Fixed #8567 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

File: pkg/etcd/etcd.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+func EtcdPath(key string) string {
+	if GlobalEtcdPrefixPath == "" || strings.HasPrefix(key, GlobalEtcdPrefixPath) {
+		return key
+	}
+	if strings.HasPrefix(key, "/") {
+		return GlobalEtcdPrefixPath + key
+	} else {
+		return GlobalEtcdPrefixPath + "/" + key
+	}
+}
```
Comment (harshavardhana) [inline pkg/etcd/etcd.go:17]: This is a bad implementation @ihostage I will send a proper implementation in a bit

**Review by harshavardhana — CHANGES_REQUESTED** [review]
thanks for taking the time to do this, but this is not correct implementation.

File: pkg/etcd/etcd.go (hunk 1 above)
Comment (ihostage) [reply pkg/etcd/etcd.go:17]: Yeeeees... I didn’t like it either. I do not know Golang and structure of MinIO very well. It was a quick resolving of the cycle of imports. I tried to create something working for initiate discussion.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#12da23f9-ea85-45d1-b79e-9f9d020be791) |
| mint-large-bucket.sh | :x: [more...](#3fd4051a-d28e-4042-bf2a-add8fd415ae3) |
<a name="12da23f9-ea85-45d1-b79e-9f9d020be791"></a>
### 8568-ddba4eb/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:30263
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 66d2c42f5b18:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-cfrrktyis1rsbv",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364575,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D9A30C7691DB23, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="3fd4051a-d28e-4042-bf2a-add8fd415ae3"></a>
### 8568-ddba4eb/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:30985
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8b9c74931f28:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 57 seconds
(4/14) Running aws-sdk-ruby tests ... done in 6 seconds
(5/14) Running awscli tests ... done in 1 minutes and 41 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 13 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 14 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 2 seconds
{
  "args": {},
  "duration": 118181,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8568-ddba4eb

**Pull request closed without merging**